R/build-redirects.R

```diff
@@ -0,0 +1,65 @@
+build_redirects <- function(pkg = ".",
+                            override = list()) {
+  pkg <- section_init(pkg, depth = 1L, override = override)
+
+  if (is.null(pkg$meta$redirects)) {
+    return(invisible())
+  }
+
+  rule("Building redirects")
+  if (is.null(pkg$meta$url)) {
+    abort(sprintf("Can't find %s.", pkgdown_field(pkg, "url")))
+  }
+
+  # after the local search PR is merged the sitemap will be actually complete
+  sitemap <- xml2::read_xml(file.path(pkg$dst_path, "sitemap.xml"))
+  paths <- xml2::xml_contents(sitemap) %>%
+    purrr::map_chr(xml2::xml_text) %>%
+    get_url_paths()
+
+  purrr::walk(
+    pkg$meta$redirects,
+    build_redirect,
+    pkg = pkg,
+    paths = paths
+  )
+}
+
+build_redirect <- function(entry, pkg, paths) {
+  new <- entry[2]
+  old <- entry[1]
+
+  # Old page must not exist
+  if (get_url_paths(old) %in% paths) {
+    abort(
+      sprintf(
+        "Must redirect an non-existing page:\nPage(s) %s from %s exist(s) in the built site.",
```

Comment:
I'm wondering whether having this is a good idea after all.
In Netlify for instance https://docs.netlify.com/routing/redirects/rewrites-proxies/#shadowing you can choose to add a redirect for actually existing pages. Maybe this should be possible in pkgdown i.e. I should remove this check.

Reply:
Yeah, that's fine with me.